### PR TITLE
Sql tips

### PR DESCRIPTION
**Pull request by jz263**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# How Has This Been Tested?

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/Main_Course/Data_Analytics/2021-08-27-SQL-tips-2.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+## 1. 引言
+
+上回Void同学主要聊了聊他对使用过的数据仓库的体验和测评。与Void"花式干饭"不同，本人在工作中主要接触的是Snowflake，以及在不同的平台里连接Snowflake写SQL，也整理一些初入数据分析常踩的坑，希望可以帮助大家绕过这些问题。
+
+## 2. SQL结构篇
+
+### 2.1 选择合适驱动表
+
+进行两表或者多表Join的时候，一般会考虑需求和性能两个方面:
+* 业务需求上：选择和结果表粒度相同的表作为驱动表：例如，如果结果表是统计某一天一群目标用户的某个行为指标，一般会选择用一张用户表的快照作为驱动表进行筛选，然后左连接相应的用户行为表。
```
Comment (Realvincentyuan) [inline docs/Main_Course/Data_Analytics/2021-08-27-SQL-tips-2.md:19]: 不同样式的元素之间要空一行

File: docs/Main_Course/Data_Analytics/2021-08-27-SQL-tips-2.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+on dept.dept_no = mgn.dept_no
+Where dept.emp_no != mgn.emp_no
+```
+
+### 2.3 善用公用表达式(CTE)
+
+公用表达式（CTE）类似编程语言中的函数，可以用于提炼出本来需要出现多次的子查询语句，有效减少代码行数。
+
+除了让查询语句变得更加清爽整洁以外，公用表达式还可以实现递归这类"高级"查询功能。按照递归三要素：
+- 终止条件
```
Comment (Realvincentyuan) [inline docs/Main_Course/Data_Analytics/2021-08-27-SQL-tips-2.md:130]: 应该要空行

**Review by Realvincentyuan — APPROVED** [review]
有些地方要空行

File: docs/Main_Course/Data_Analytics/2021-08-27-SQL-tips-2.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+上回Void同学主要聊了聊他对使用过的数据仓库的体验和测评。与Void"花式干饭"不同，本人在工作中主要接触的是Snowflake，以及在不同的平台里连接Snowflake写SQL，也整理一些初入数据分析常踩的坑，希望可以帮助大家绕过这些问题。
+
+## 2. SQL结构篇
+
+### 2.1 选择合适驱动表
+
+进行两表或者多表Join的时候，一般会考虑需求和性能两个方面:
+* 业务需求上：选择和结果表粒度相同的表作为驱动表：例如，如果结果表是统计某一天一群目标用户的某个行为指标，一般会选择用一张用户表的快照作为驱动表进行筛选，然后左连接相应的用户行为表。
+* 性能上：选择小表作为驱动表连接大表，对大表建立索引：如果两张需要Join的表记录数量差异很大，一般会把小表放在前面。因为在表连接时，只有右连接表里的索引会生效，且索引的在大表中效率提升更明显。
```
Comment (jhqian0909) [inline docs/Main_Course/Data_Analytics/2021-08-27-SQL-tips-2.md:20]: 索引在大表中

File: docs/Main_Course/Data_Analytics/2021-08-27-SQL-tips-2.md [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+| d001    | 10001  | 1996-08-03 | 9999-01-01 |
+| d002    | 10003  | 1990-08-05 | 9999-01-01 |
+
+我们想要找到所有员工对应的经理，如果员工本身是经理的话则不显示，以上例子如下：
+
+| emp_no | manger |
+|--------|--------|
+| 10002  | 10001 |
+
+注意第一张表种，emp_no是有重的，不去重直接连接的话会出现两条记录:
```
Comment (jhqian0909) [inline docs/Main_Course/Data_Analytics/2021-08-27-SQL-tips-2.md:53]: 表中

File: docs/Main_Course/Data_Analytics/2021-08-27-SQL-tips-2.md (hunk 3 above)
Comment (jhqian0909) [reply docs/Main_Course/Data_Analytics/2021-08-27-SQL-tips-2.md:20]: 只有右连接表里的索引生效这个只是针对snowflake吗

Comment (jz263) [reply docs/Main_Course/Data_Analytics/2021-08-27-SQL-tips-2.md:20]: mysql也是，原理就是会循环抽取生成的左表（驱动表）的一条记录，然后和右表匹配，这个时候如果正好是通过右表索引字段匹配的话，索引就会生效

File: docs/Main_Course/Data_Analytics/2021-08-27-SQL-tips-2.md (hunk 1 above)
Comment (jz263) [reply docs/Main_Course/Data_Analytics/2021-08-27-SQL-tips-2.md:20]: 已修改

File: docs/Main_Course/Data_Analytics/2021-08-27-SQL-tips-2.md (hunk 2 above)
Comment (jz263) [reply docs/Main_Course/Data_Analytics/2021-08-27-SQL-tips-2.md:132]: 已修改

File: docs/Main_Course/Data_Analytics/2021-08-27-SQL-tips-2.md (hunk 4 above)
Comment (jz263) [reply docs/Main_Course/Data_Analytics/2021-08-27-SQL-tips-2.md]: 已修改

**Pull request merged**